network/codec/cbor/decoder.go

```diff
@@ -10,29 +10,36 @@ import (
 	_ "github.com/onflow/flow-go/utils/binstat"
 )
 
-// Decoder implements a stream decoder for JSON.
+// Decoder implements a stream decoder for CBOR.
 type Decoder struct {
 	dec *cbor.Decoder
 }
 
-// Decode will decode the next JSON value from the stream.
+// Decode will decode the next CBOR value from the stream.
 func (d *Decoder) Decode() (interface{}, error) {
 
-	// decode the next envelope
+	// read from stream and extract code
 	var data []byte
-	//bs := binstat.EnterTime(binstat.BinNet + ":strm>1(cbor)")
+	//bs1 := binstat.EnterTime(binstat.BinNet + ":strm>1(cbor)iowriter2payload2envelope")
 	err := d.dec.Decode(data)
-	//binstat.LeaveVal(bs, int64(len(data)))
+	//binstat.LeaveVal(bs1, int64(len(data)))
 	if err != nil {
```

Comment:
```suggestion
	if err != nil || len(data) == 0 {
```

Reply:
@vishalchangrani thanks for the comment. I added the extra `len(data) == 0`. Thanks for spotting this :-)